### PR TITLE
Fix result onerror

### PR DESCRIPTION
**Pull request by OmarWKH**
Fixes the onError part of https://github.com/czue/celery-progress/issues/66.

Also makes make onResultDefault an instance method following the discussion in https://github.com/czue/celery-progress/pull/62.

### CODE REVIEW

**Comment by czue** [comment]
I haven't been following that thread closely (sorry, life is busy right now!) so happy to defer to you and @EJH2 on this one

**Comment by czue** [comment]
@EJH2 any thoughts on this? I can also take a closer look sometime soon if you don't have time

**Review by EJH2 — APPROVED** [review]
Sorry about this, I got wrapped up in another project and school and didn't have time to look at this. Changes look good and elegant for the fix, making onResult static was going to need to happen anyways so glad that got added too. Should be good to go with @czue's review when he has the time.

**Pull request merged**
